### DISCUSSION TITLE
How are you actually meant to implement custom Item components?

### DISCUSSION THREAD

**Discussion by angusd3v** [adobe/react-spectrum#3483] [Q&A] [answered]
Alternative title: Why is `Item` generic if I can't actually provide any of my own props to it?

When talking about collection components, the docs talk about the usefulness of JSX for defining collections.

> React Spectrum implements a JSX-based API for defining collections. This allows an intuitive way to provide items, which can contain rich rendering, and various options as props. Building hierarchical collections, e.g. sections, or a tree of items is also very natural in JSX.
> `<Item>` and `<Section>` only define the data for the items and sections, not the rendering, visual appearance, or behavior. This is up to the individual collection component (e.g. Menu or ListBox) to implement.

However, it isn't clear to me how I'm actually meant to use what appears to be the core collection component (`Item`) in this way. `Item` is generic, but the `T` just gets essentially thrown away:

![image](https://user-images.githubusercontent.com/52683145/188531950-94a028d0-790c-4649-90ab-3ddcd5f43708.png)

Say if I want to create my own components using collections, is there actually a way to do this with Item?

e.g. a component which renders a bottom nav bar with tabs that have an icon and label, and selecting one of the tab items puts the children of that item into a bottom sheet

```
interface BottomNavItemProps {
  children: React.ReactNode;
  icon: ReactSVGComponent;
  title: string;
}

<BottomNav>
  <Item<BottomNavItemProps> icon={MySvg} id="tab1" title="Tab title">
    <ContentsOfTheTab>
  </Item>
  <Item<BottomNavItemProps> icon={MySvg} id="tab2" title="Tab title">
    <ContentsOfTheTab>
  </Item>
</BottomNav>
```

Is it intentional that passing the generic `BottomNavItemProps` doesn't actually allow you to pass those props to the `Item`? I just get errors saying I can't pass `icon`, which really diminishes my perceived usefulness of Item. 

It appears that adding my own `getCollectionNode` generator function might be the way that I'm meant to do this, but to me that seems much less readable/grokkable to my team members, and it also involves some messy type workarounds to hide `getCollectionNode` from the visible types.

**Comment by LFDanLu** [comment, marked as answer]
You could potentially override the `Item`'s props in a re-export like we do here for [Column](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/table/src/index.ts#L20). Your additional props would then be available in the node's `props`. This is all a bit messy since `Item` wasn't really meant to be extended.

The team has been experimenting with allowing `Item` wrappers but that work is still in progress
